common/src/main/java/com/wynntils/features/wynntils/BetaWarningFeature.java

```diff
@@ -39,15 +38,19 @@ public void onDisplayNameChange(PlayerInfoEvent.PlayerDisplayNameChangeEvent e)
         if (warnType == WarnType.NONE) return;
 
         McUtils.sendMessageToClient(warnType.getWarning());
+
+        // Send a link to our Discord server for beta builds, we can't include the link in the message as Urls have
+        // not been loaded by the time the enum is initialised
+        if (warnType == WarnType.RELEASE) {
+            Handlers.Command.sendCommandImmediately("wynntils discord");
+        }
         warnType = WarnType.NONE;
     }
 
     private enum WarnType {
         NONE(Component.empty()),
         BETA(Component.translatable("feature.wynntils.betaWarning.usingBetaOnNormalServer")),
-        RELEASE(Component.translatable(
-                "feature.wynntils.betaWarning.usingReleaseOnBetaServer",
-                Managers.Url.getUrl(UrlId.LINK_WYNNTILS_DISCORD_INVITE)));
+        RELEASE(Component.translatable("feature.wynntils.betaWarning.usingReleaseOnBetaServer"));
```

Comment:
What if you made this be "supplied"? So just make the enum contain a Supplier<Component>. I think that should work better and is much cleaner. Make sure to include a line about it in comments.